src/main/java/org/medicmobile/webapp/mobile/ChtExternalAppLauncher.java

```diff
@@ -0,0 +1,378 @@
+				.setPackageName(chtExternalApp.getPackageName())
+				.setUri(chtExternalApp.getUri())
+				.setFlags(chtExternalApp.getFlags())
+				.setType(chtExternalApp.getType())
+				.build();
+	}
+
+	static String processResponse(Intent intent, Activity context) {
+		try {
+			if (intent == null || intent.getExtras() == null) {
+				return makeJavaScript(null);
+			}
+
+			JSONObject json = parseBundleToJson(intent.getExtras(), context);
+			return makeJavaScript(json);
+
+		} catch (Exception exception) {
+			error(exception, "ChtExternalAppLauncher :: Problem serialising the intent response");
+			return safeFormat("console.error('Problem serialising the intent response: %s')", exception);
+		}
+	}
+
+	//> PRIVATE
+
+	private static String makeJavaScript(Object response) {
+		String javaScript = "try {" +
+				"const api = window.CHTCore.AndroidApi;" +
+				"if (api && api.resolveCHTExternalAppResponse) {" +
+				"  api.resolveCHTExternalAppResponse(%s);" +
+				"}" +
+				"} catch (error) { " +
+				"  console.error('ChtExternalAppLauncher :: Error on sending intent response to CHT-Core webapp', error);" +
+				"}";
+
+		return safeFormat(javaScript, response);
+	}
+
+	private static JSONObject parseBundleToJson(Bundle bundle, Activity context) {
+		try {
+			JSONObject json = new JSONObject();
+			bundle
+					.keySet()
+					.forEach(key -> setValueInJson(key, json, bundle, context));
+			return json;
+
+		} catch (Exception exception) {
+			error(exception, "ChtExternalAppLauncher :: Problem parsing bundle to json. Bundle=%s", bundle);
+		}
+
+		return null;
+	}
+
+	private static void setValueInJson(String key, JSONObject json, Bundle bundle, Activity context) {
+		try {
+			Object value = bundle.get(key);
+
+			if (value instanceof Bitmap) {
+				json.put(key, parseBitmapImageToBase64((Bitmap) value, context));
+				return;
+			}
+
+			if (value instanceof Bundle) {
+				json.put(key, parseBundleToJson((Bundle) value, context));
+				return;
+			}
+
+			if (isBundleList(value)) {
+				JSONArray jsonArray = ((List<Bundle>) value)
+						.stream()
+						.map(item -> parseBundleToJson(item, context))
+						.collect(Collector.of(JSONArray::new, JSONArray::put, JSONArray::put));
+
+				json.put(key, jsonArray);
+				return;
+			}
+
+			if (isStorageUri(value)) {
+				json.put(key, getImageFromStoragePath((String) value, context));
+				return;
+			}
+
+			json.put(key, JSONObject.wrap(value));
+
+		} catch (Exception exception) {
+			error(exception, "ChtExternalAppLauncher :: Problem parsing bundle to json. Key=%s, Bundle=%s", key, bundle);
+		}
+	}
+
+	private static boolean isBundleList(Object value) {
+		return value instanceof List
+				&& ((List<?>) value).size() > 0
+				&& ((List<?>) value).get(0) instanceof Bundle;
+	}
+
+	private static boolean isStorageUri(Object value) {
+		return value instanceof String
+				&& (((String) value).contains("file://") || ((String) value).contains("content://"));
```

Comment:
I think `value.contains("file://")` doesn't work (Android 9).. Maybe checking by extension `contains(".jpg")` ☹️ ? 
gotta find something better.

Reply:
I've fixed it, validated with Android 9 and 11 devices.